src/hand.rs

```diff
@@ -0,0 +1,100 @@
+use super::bag::TileBag;
+
+pub struct Hands {
+    hands: Vec<Vec<char>>,
+    bag: TileBag,
+}
+
+impl Hands {
+    pub fn new(player_count: usize, capacity: usize, mut bag: TileBag) -> Self {
+        let mut hands = Vec::new();
+        for player in 0..player_count {
+            hands.push(Vec::new());
```

Comment:
To save some allocations :) 

```suggestion
            hands.push(Vec::with_capacity(capacity));
```

Reply:
Oooh nice!